src/k8s/vmBuilder.js

```diff
@@ -282,3 +282,86 @@ export const getDataVolumeTemplateSpec = (storage, dvSource) => {
     },
   };
 };
+
+export const assignBootOrderIndex = (vm, currDevBootOrder = -1) => {
+  let bootOrder = currDevBootOrder;
+  if (currDevBootOrder !== BOOT_ORDER_FIRST) {
+    bootOrder = getNextAvailableBootOrderIndex(vm);
+  }
+  return bootOrder;
+};
+
+const getNextAvailableBootOrderIndex = vm => {
+  const disks = getDisks(vm);
+  const nics = getInterfaces(vm);
+  let largestIdx = -1;
+
+  disks.forEach(disk => {
+    const bootOrder = get(disk, 'bootOrder', -1);
+    largestIdx = bootOrder > largestIdx ? bootOrder : largestIdx;
+  });
+
+  nics.forEach(nic => {
+    const bootOrder = get(nic, 'bootOrder', -1);
+    largestIdx = bootOrder > largestIdx ? bootOrder : largestIdx;
+  });
+
+  // assigned indexes start at two as the first index is assigned directly by the user
+  return largestIdx !== -1 ? largestIdx + 1 : BOOT_ORDER_SECOND;
+};
+
+export const sequentializeBootOrderIndexes = vm => {
+  const sortedBootableDevices = getBootableDevicesInOrder(vm);
+
+  let index;
+  for (index = 0; index <= sortedBootableDevices.length; index++) {
```

Comment:
.forEach((device, index) => {}) is more readable IMO

Reply:
Done!